x-pack/solutions/security/plugins/security_solution/common/api/detection_engine/prebuilt_rules/get_prebuilt_rule_base_version/get_prebuilt_rule_base_version_route.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod';
+import type { RuleResponse } from '../../model/rule_schema/rule_schemas.gen';
+import type { PartialRuleDiff } from '../model';
+
+export type GetPrebuiltRuleBaseVersionRequest = z.infer<typeof GetPrebuiltRuleBaseVersionRequest>;
+export const GetPrebuiltRuleBaseVersionRequest = z.object({
+  id: z.string(),
+});
+
+export interface GetPrebuiltRuleBaseVersionResponseBody {
+  /** The base version of the rule */
+  base_version: RuleResponse;
+
+  /** The current version of the rule */
+  current_version: RuleResponse;
+
+  /** The resulting diff between the base and current versions of the rule */
+  diff: PartialRuleDiff;
```

Comment:
As an improvement you could return `fieldsDiff: Partial<RuleFieldsDiff>` and add the necessary fallback fields for example in `x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/base_version_diff/base_version_flyout.tsx` or in the upstream data fetching hook.